src/Command/QueueCommand.php

```diff
@@ -73,23 +72,23 @@ protected function configure()
         try {
             // Transform into something for gearman.
             $entity = $this->transformer->transform($item, $this->serializedTransform);
-        } catch (BadResponse $e) {
-            // We got a 404 or server error.
-            $this->logger->error("{$this->getName()}: Item does not exist in API: {$item->getType()} ({$item->getId()})", [
-                'exception' => $e,
-                'item' => $item,
-            ]);
-            // Remove from queue.
-            $this->queue->commit($item);
         } catch (Throwable $e) {
-            // Unknown error.
-            $this->logger->error("{$this->getName()}: There was an unknown problem importing {$item->getType()} ({$item->getId()})", [
-                'exception' => $e,
-                'item' => $item,
-            ]);
-            $this->monitoring->recordException($e, "Error in importing {$item->getType()} {$item->getId()}");
-            // Remove from queue.
-            $this->queue->commit($item);
+            if ($e instanceof BadResponse && in_array($e->getResponse()->getStatusCode(), [404, 410])) {
+                $this->logger->error("{$this->getName()}: Item does not exist in API: {$item->getType()} ({$item->getId()})", [
+                    'exception' => $e,
+                    'item' => $item,
+                ]);
+                // Remove from queue.
+                $this->queue->commit($item);
+            } else {
+                // Unknown error.
+                $this->logger->error("{$this->getName()}: There was an unknown problem importing {$item->getType()} ({$item->getId()})", [
+                    'exception' => $e,
+                    'item' => $item,
+                ]);
+                $this->monitoring->recordException($e, "Error in importing {$item->getType()} {$item->getId()}");
+                $this->release($item);
```

Comment:
If we didn't release the item, it will go back to the queue but after the visibility timeout (60 seconds usually). This could avoid overloading when continuously reprocessing events. Exponential backoff is not supported in SQS without unreliable hacks.

Reply:
That seems the implementation of release(), but I'd rather not call `release()` here to postpone the moment in which the message will become visible again.

Reply:
Would break with other implementations? Should `SqsWatchableQueue::release()` do nothing?

Reply:
Comparing https://www.rabbitmq.com/tutorials/tutorial-two-python.html

```
There aren't any message timeouts; RabbitMQ will redeliver the message when the consumer dies. It's fine even if processing a message takes a very, very long time.
```

Other implementations like ActiveMQ seem similar as they are connection-oriented and they use that to detect the consumer crashing. A more conservative solution then would be to `sleep(X)` and then release, which slows down the consumer and has no backoff, but is portable.

Reply:
The `VisibilityTimeout` in `SqsWatchableQueue::release()` could be set to `X` rather than the current 0, using `ApproximateReceiveCount` from `receiveMessage` to implement a backoff?

Reply:
:+1: approximation is good enough

Reply:
Added.